ethcore/trace/Cargo.toml

```diff
@@ -6,7 +6,6 @@ authors = ["Parity Technologies <admin@parity.io>"]
 edition = "2018"
 
 [dependencies]
-common-types = { path = "../types"}
 ethcore-blockchain = { path = "../blockchain" }
```

Comment:
Hmm, so we still can't get rid of the blockchain depenency? Then I don't think it makes much sense.

Reply:
No, we cannot: https://github.com/paritytech/parity-ethereum/blob/44cc442d12348641395531980bf913659561d012/ethcore/trace/src/db.rs#L72

Reply:
`dyn BlockChainWithExtras`? :)

Reply:
`dyn BlockChainWithExtrasAndExtraMayo` ;)

Sure, we can work on disentangling things further, but I think we can get more bang for our effort elsewhere at this time. I'm fine adding an issue to make `trace` independent of `blockchain` but for the time being I think we should pick the lesser of two evils, so the question is: which is worst, this PR or the current master? Personally I think they both smell a little so I don't really care which it is. @tomusdrw seem to feel this PR is worse. What is your opinion @debris, merge or close?

Reply:
Since we depend on `blockchain` the wrapper type is unecessary since we can write the implementation right next to the `DatabaseExtras` definition. I'd just add this in `traces`:
```rust
impl<T: BlockProvider> trace::DatabaseExtras for T {
	fn block_hash(&self, block_number: BlockNumber) -> Option<H256> {
		BlockProvider::block_hash(&self, block_number)
	}

	fn transaction_hash(&self, block_number: BlockNumber, tx_position: usize) -> Option<H256> {
		self.block_hash(block_number)
			.and_then(|block_hash| {
				let tx_address = TransactionAddress {
					block_hash,
					index: tx_position
				};
				self.transaction(&tx_address)
			})
			.map(|tx| tx.hash())
	}
}
```
and be done with it.